smartcontractkit/external-adapters-js
# Feature/ea env var context

## Closes [31905](https://app.shortcut.com/chainlinklabs/story/31905/ea-environment-variables-context)

## Description
Add `baseEnvDefaults` to consolidate env var defaults, and extend adapter context to support default overrides with `envDefaultOverrides`. If an env var is defined then that value is used, but if not then the adapter override default is used, followed by the base env default if no override is given. Override defaults have been added for `ncfx` and `onchain-gas` for the `WS_ENABLED` env var as an example.

## Changes

- Consolidate env var defaults in `baseEnvDefaults`
- Replace all `process.env` usage with `getEnv` for consistency
- Add `envDefaultOverrides` to adapter context passed into `expose` to allow adapter-specific env var default overrides
- Coalesce the env var base and adapter override defaults in `getEnv`, with adapter override defaults taking precedent over base defaults
- Add default overrides for `WS_ENABLED` in `ncfx` and `onchain-gas` adapters to demonstrate usage
- Add default override for `CACHE_ENABLED` in `layer2-sequencer-health` adapter

## Steps to Test
1. Comment out `WS_ENABLED: 'true',` from `packages/sources/ncfx/src/config/envDefaultOverrides.ts`.
2. Export proper API_USERNAME and API_PASSWORD for the `ncfx` adapter.
3. Build and start the `ncfx` adapter.
4. Run a request like `{"jobRunId": 1, "data": {"base": "ETH", "quote": "USD"}}` and verify the result is a `500` error with the following message:
```
"The NCFX adapter does not support making HTTP requests. Make sure WS is enabled in the adapter configuration."
 ```
5. Uncomment the `WS_ENABLED: 'true',` line.
6. Re-build and re-start the `ncfx` adapter.
7. Run the same request and verify the result is a successful response.

## Quality Assurance
